setup.py

```diff
@@ -32,11 +33,19 @@ def load_requirements(file_list: Optional[Union[str, List[str]]] = None) -> List
         file_list = [file_list]
     requirements: List[str] = []
     for file in file_list:
+        path = Path(file)
+        file_dir = path.parent.resolve()
         with open(file, encoding="utf-8") as f:
-            requirements.extend(f.readlines())
-    requirements = [
-        req for req in requirements if not req.startswith("#") and not req.startswith("-")
-    ]
+            raw = f.read().splitlines()
+            requirements.extend([r for r in raw if not r.startswith("#") and not r.startswith("-")])
+            recursive_reqs = [r for r in raw if r.startswith("-r")]
+            if recursive_reqs:
+                filenames = []
+                for recursive_req in recursive_reqs:
+                    file_spec = recursive_req.split()[-1]
+                    file_path = Path(file_dir) / file_spec
+                    filenames.append(str(file_path.resolve()))
+                requirements.extend(load_requirements(file_list=filenames))
```

Comment:
We don't need to keep the file open after we've read it, so this block can be moved out of the `with` clause.